engine/flink/components/table/src/main/scala/pl/touk/nussknacker/engine/flink/table/source/FlinkMiniClusterDataGenerator.scala

```diff
@@ -0,0 +1,132 @@
+package pl.touk.nussknacker.engine.flink.table.source
+
+import cats.implicits._
+import com.typesafe.scalalogging.LazyLogging
+import io.circe.parser.parse
+import org.apache.commons.io.FileUtils
+import org.apache.flink.configuration.{Configuration, CoreOptions, PipelineOptions}
+import org.apache.flink.table.api.{EnvironmentSettings, Schema, TableDescriptor, TableEnvironment}
+import pl.touk.nussknacker.engine.api.test.{TestData, TestRecord}
+import pl.touk.nussknacker.engine.flink.table.source.FlinkMiniClusterDataGenerator._
+import pl.touk.nussknacker.engine.util.ThreadUtils
+
+import java.net.URL
+import java.nio.charset.StandardCharsets
+import java.nio.file.{DirectoryStream, Files, Path}
+import scala.jdk.CollectionConverters._
+import scala.util.{Failure, Success, Try}
+
+class FlinkMiniClusterDataGenerator(flinkTableSchema: Schema) extends LazyLogging {
+
+  // TODO: check the minicluster releases memory properly and if not, refactor to reuse one minicluster per all usages
```

Comment:
I see that at least an executor is created. I see no method to release the resources tho :(
But maybe we can create the `env` in the constructor of this class? At least each run of test generation within the same instance of node, won't create the mini cluster. 

(obviously, in the future, we could do as it's described in the comment)

Reply:
Ok, when I implemented this, I checked that the TableSource and FlinkMiniClusterDataGenerator are initialized each time the test data generation is called :/ So even when we reuse the minicluster on FlinkMiniClusterDataGenerator or even component level, it doesn't matter :/ Outside of some larger refactor I don't see a solution